consensus/ibft/ibft.go

```diff
@@ -448,10 +432,52 @@ func (i *Ibft) buildBlock(snap *Snapshot, parent *types.Header) (*types.Block, e
 	// is sealed after all the committed seals
 	block.Header.ComputeHash()
 
-	i.logger.Info("build block", "number", header.Number, "txns", len(txns))
+	i.logger.Info("build block", "number", header.Number, "txPoolLength", len(txns))
 	return block, nil
 }
 
+type transitionInterface interface {
+	Write(txn *types.Transaction) error
+}
+
+func (i *Ibft) writeTransactions(gasLimit uint64, transition transitionInterface) []*types.Transaction {
+	txns := []*types.Transaction{}
+	returnTxnFuncs := []func(){}
+	for {
+		txn, retTxnFn := i.txpool.Pop()
+		if txn == nil {
+			break
+		}
+
+		if txn.ExceedsBlockGasLimit(gasLimit) {
+			i.logger.Error(fmt.Sprintf("failed to write transaction: %v", state.ErrBlockLimitExceeded))
+			i.txpool.DecreaseAccountNonce(txn)
+			continue
+		}
+
+		if err := transition.Write(txn); err != nil {
+			if appErr, ok := err.(*state.GasLimitReachedTransitionApplicationError); ok && appErr.IsRecoverable {
```

Comment:
Wondering whether we should check appErr.IsRecoverable is true because it's IsRecoverable 
 in case of GasLimitReachedTransitionApplicationError

Reply:
I also think this is not necessary, i was intending to remove this but obviously forgot. Tnx for the notice

Reply:
[changed](https://github.com/0xPolygon/polygon-sdk/pull/237/commits/ecdc987abb40a81960d8dbbf35b60fe59ef78c53)